src/scripts/Update.ts

```diff
@@ -2309,6 +2315,25 @@ class Update implements Saveable {
         }
     }
 
+    // Use setBattleState as 0 or 1 to manipulate battles to what status they should be based on related questline progress.
+    static fixTempBattleState = (saveData, battleName: string, setBattleState: number, questLineName: string, questStep: number) => {
+        const ql = saveData.quests.questLines.find((q) => q.name === questLineName);
+
+        if (setBattleState === 1) {
+            // set to complete if related questline/step is completed
+            if (ql?.state === 2 || ql?.quest > questStep) {
+                saveData.statistics.temporaryBattleDefeated[GameConstants.getTemporaryBattlesIndex(battleName)] = 1;
+            }
+        }
+
+        if (setBattleState === 0) {
+            // set to not complete if related questline/step isn't complete
+            if (ql?.state < 2 && ql?.quest <= questStep) {
+                saveData.statistics.temporaryBattleDefeated[GameConstants.getTemporaryBattlesIndex(battleName)] = 0;
```

Comment:
Same as above with magic numbers

Reply:
Changed battleNames to the index and added a return if `ql` is undefined